prebid-mobile/pbm-api/ios/pbm-plugin-renderer.md

```diff
@@ -0,0 +1,189 @@
+
+Take the example on the image below where a BannerView will have its ad view transposed accordingly to the Plugin Renderer status. The inner ad view is handled totally under the hood from the app owner point of view, what makes unnecessary any change on the BannerView loading or initialization.
+
+{: .alert.alert-info :}
+In case of Interstitial ad this is just inflated in the foreground regardless the view hierarchy.
+
+![Plugin Renderer big picture](/assets/images/prebid-mobile/prebid-plugin-renderer-ad-view-injection.png)
+
+### Setup
+
+* Provide your Prebid Bidder Adapter
+  * [Go integration](/prebid-server/developers/add-new-bidder-go.html)
+  * [Java integration](/prebid-server/developers/add-new-bidder-java.html)
+* Create your implementation from the interface `PrebidMobilePluginRenderer`
+* Initialise your Plugin Renderer before starting to request ads
+* Take advantage of the Plugin Renderer fields
+
+{: .alert.alert-info :}
+Please notice that all implementation on mobile related to the Plugin Renderer should be provided externally, not in the PBM SDK itself. For instance, an app owner or third party SDK would implement it and initialise it on their own context.
+
+___
+
+#### Create your implementation from the interface PrebidMobilePluginRenderer
+
+```swift
+public class SampleCustomRenderer: NSObject, PrebidMobilePluginRenderer {
+    
+    public let name = "SampleCustomRenderer"
+    
+    public let version = "1.0.0"
+    
+    public var data: [AnyHashable: Any]? = nil
+    
+    private var adViewManager: PBMAdViewManager?
+    
+    public func isSupportRendering(for format: AdFormat?) -> Bool {}
+   
+    public func setupBid(_ bid: Bid, adConfiguration: AdUnitConfig, connection: PrebidServerConnectionProtocol) {}
+    
+    public func createBannerAdView(with frame: CGRect, bid: Bid, adConfiguration: AdUnitConfig,
+                                   connection: PrebidServerConnectionProtocol, adViewDelegate: (any PBMAdViewDelegate)?) {
+        // TODO "Handle bid response as you want and display your banner ad"
+    }
+
+    public func createInterstitialController(bid: Bid, adConfiguration: AdUnitConfig, connection: PrebidServerConnectionProtocol,
+                                             adViewManagerDelegate adViewDelegate: InterstitialController?, videoControlsConfig: VideoControlsConfiguration?) {
+        // TODO "Handle bid response as you want and display your interstitial ad"
+    }
+}
+
+```
+
+#### Initialise your Plugin Renderer before starting to request ads and unregister your plugin once it is done
+
+```swift
+class CustomRendererBannerController: NSObject, AdaptedController, PrebidConfigurableBannerController, BannerViewDelegate {
+    
+    required init(rootController: AdapterViewController) {
+        super.init()
+        self.rootController = rootController
+        Prebid.registerPluginRenderer(sampleCustomRenderer)
```

Comment:
Does it make sense to register the plugin during the application launch step right after the Prebid SDK initialization?  In this case, publishers will not have to register the plugin in each needed place in the app.